lib/ddtrace/span.rb

```diff
@@ -68,7 +68,11 @@ def get_tag(key)
     # Set the given key / value metric pair on the span. Keys must be string.
     # Values must be floating point numbers.
     def set_metric(key, value)
+      # enforce that the value is a floating point number
+      value = Float(value)
       @metrics[key] = value
+    rescue StandardError => e
```

Comment:
Note that a cast will raise an `ArgumentError`.
A cleaner way would be to check if `value.is_a?(Integer) || value.is_a?(Float)`.